PUSH_TO_FILE.md

```diff
@@ -0,0 +1,289 @@
+| conjur.org/secret-file-path.{secret-group}    | This configuration is not available with environmental variables | Relative path for secrets file or directory to be written. This path is assumed to be relative to the respective mount path for the shared secrets volume for each container (for the Secrets Provider container, the mount path is `/conjur/secrets/`).  The file path can be either a directory (any value ending in `/`) or a path that includes a file name. For directory paths, a default file name of the form `{secret-group}.{secret-file-format}` is assumed. Since this is a relative path, values beginning with `/` will be rejected and will cause the Secrets Provider to abort. |
+| conjur.org/secret-file-format.{secret-group}  | This configuration is not available with environmental variables | Allowed values:yaml (default) |
+
+
+## Volume mounts
+When the Secrets Provider is configured for file mode, as described above, it will 
+write secrets to file(s) in an volume that is shared with the application container. 
+The volumes required for this mode are as follows:
+* `conjur-secrets`: An `emptydir` volumed mounted to both the application container 
+and Secrets Provider.  Secrets fetched from Conjur are written here.
+* `podinfo`: A volume mounted to just Secrets Provider containing pod annotations and 
+metadata from the Downward API.
+
+Below is sample YAML defining the two volumes:
+```
+volumes:
+  - name: podinfo
+  downwardAPI:
+    items:
+      - path: "annotations"
+        fieldRef:
+          fieldPath: metadata.annotations
+  - name: conjur-secrets
+    emptyDir:
+      medium: Memory
+```
+Below is sample volume mounts for the Secrets Provider init container:
+```
+volumeMounts:
+  - name: podinfo
+    mountPath: /conjur/podinfo
+  - name: conjur-secrets
+    mountPath: /conjur/secrets
+```
+
+Below is sample volume mount for the target application:
+```
+volumeMounts:
+  - name: conjur-secrets
+    mountPath: /opt/secrets/conjur
+    readOnly: true
+```
+
+By default, the Secrets Provider will create secrets files with the following file attributes:
+
+|  Attribute  |       Value        | Notes  |
+| ----------- | ------------------ | ------ |
+| User        | `secrets-provider` |        |
+| Group       | `root`             | OpenShift requires that any files/directories that are shared between containers in a Pod must use a GID of 0 (i.e. GID for the root group) The Secrets Provider uses a GID of 0 for secrets files even for non-OpenShift platforms, for simplicity.       |
+| UID         | `777`              |        |
+| GID         | `0`                |        |
+| Permissions | `rw-rw-r--`        | As shown in the table table, the Secrets Provider will create secrets files that are world readable. This means that the files will be readable from any other container in the same Pod that mounts the Conjur secrets shared Volume.       |
+
```

Comment:
Since we're saying this config is "By default", we probably want to mention that if desired, these can be overridden by defining a PodSecurityContext (https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#podsecuritycontext-v1-core) for the application Pod.

For example, to have containers run as the `nobody` user (UID 65534) and `nobody` group (GID 64434), and have secrets files created with the corresponding GID, the Pod SecurityContext would be:
```
    securityContext:
      runAsUser: 65534
      runAsGroup: 65534
      runAsNonRoot: true
      fsGroup: 65534
```